Core/src/main/java/com/paypal/android/core/API.kt

```diff
@@ -1,13 +1,30 @@
 package com.paypal.android.core
 
+import android.content.Context
+import android.util.Log
+import androidx.annotation.VisibleForTesting
+import com.paypal.android.core.analytics.AnalyticsEventData
+import com.paypal.android.core.analytics.DeviceInspector
+import java.util.UUID
+
 class API internal constructor(
     private val configuration: CoreConfig,
+    private val applicationContext: Context,
```

Comment:
`API` doesnt need the `applicationContext`, why dont we pass in `DeviceInspector` constructor?

Reply:
It's used in the `sendAnalyticsEvent()` method. It was better for testability.

Reply:
im just wondering why can it not live inside `DeviceInspector`, where it is actually used, does it affect testability that much? Feels like an uneccessary variable inside `API`